app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/NotesModule.kt

```diff
@@ -2,17 +2,18 @@ package de.westnordost.streetcomplete.data.osmnotes
 
 import android.content.Context
 import de.westnordost.streetcomplete.ApplicationConstants
-import org.koin.core.qualifier.named
+import okio.Path.Companion.toOkioPath
 import org.koin.dsl.module
-import java.io.File
 
 val notesModule = module {
-    factory(named("AvatarsCacheDirectory")) { File(get<Context>().cacheDir, ApplicationConstants.AVATARS_CACHE_DIRECTORY) }
-    factory { AvatarsDownloader(get(), get(), get(named("AvatarsCacheDirectory"))) }
+    factory {
+        val avatarCacheDir = get<Context>().cacheDir.toOkioPath().resolve(ApplicationConstants.AVATARS_CACHE_DIRECTORY)
+        AvatarsDownloader(get(), get(), get(), avatarCacheDir)
+    }
```

Comment:
It seems unclean to me to remove the `AvatarsCacheDirectory` as injectable dependency. (As you likely noticed,) this leads to application components that have nothing to do with download of avatar images (and should not be able to trigger an avatar download) to depend on `AvatarDownloader`.